src/schema/posts.ts

```diff
@@ -740,6 +751,28 @@ export const resolvers: IResolvers<any, Context> = {
       }
       return { _: true };
     },
+    deleteSharedPost: async (
```

Comment:
We already have a mutation for deleting a post. I'd suggest streamlining everything to the existing function.
Everything is a post but depending on a type we have a different logic, I don't want to separate the mutations

Reply:
The `deletePost` requires a direct `moderator` role from a user. 
Schema:
```
deletePost(
      """
      Id of the post to delete
      """
      id: ID
    ): EmptyResponse @auth(requires: [MODERATOR])
```

Not sure what we will try to reuse from the mutation below since it is just a direct call to the repository:
```
deletePost: async (
      source,
      { id }: { id: string },
      ctx: Context,
    ): Promise<GQLEmptyResponse> => {
      await ctx.getRepository(Post).update({ id }, { deleted: true });
      return { _: true };
    },
```

Reply:
It's not about reusing the logic, it's about making it easy for the client. Client knows it needs to delete post so there is a single mutation for it. Also from maintainer point of view, we know there's only one endpoint to delete a post

Reply:
I just think endpoints that require higher access levels should be separate for security's sake. But we can merge them if necessary.

What do you think of removing the `requires: [MODERATOR]` and then checking it at the JS level when the user is not a moderator and the post is an article?

Reply:
Yes makes sense :)